File: frontend/src/api/monitoring.ts
import { get } from './client'

export interface VendorLinkage {
  id: string
  vendor_code: string
  legal_name: string
}

export interface InvoiceLinkage {
  id: string
  invoice_number: string
}

export interface OutstandingInvoiceResponse {
  id: string
  invoice_number: string
  status: string
  total_amount: number
  paid_amount: number
  outstanding_amount: number
  aging_days: number
  is_overdue: boolean
  vendor: VendorLinkage
}

export interface PendingPaymentResponse {
  id: string
  payment_reference: string
  amount: number
  status: string
  aging_days: number
  invoice: InvoiceLinkage
  vendor: VendorLinkage
}

export interface UnsettledPaymentResponse {
  id: string
  payment_reference: string
  amount: number
  settlement_total: number
  unsettled_amount: number
  invoice: InvoiceLinkage
  vendor: VendorLinkage
}

export interface PendingSettlementResponse {
  id: string
  settlement_reference: string
  amount: number
  status: string
  aging_days: number
  vendor: VendorLinkage
  invoice: InvoiceLinkage
}

export interface FinancialSummaryResponse {
  total_outstanding_invoices: number
  total_pending_payments: number
  total_unsettled_liabilities: number
  total_pending_settlements: number
}

export const monitoringApi = {
  getOutstandingInvoices: () =>
    get<OutstandingInvoiceResponse[]>('/monitoring/outstanding-invoices'),

  getPendingPayments: () =>
    get<PendingPaymentResponse[]>('/monitoring/pending-payments'),

  getUnsettledPayments: () =>
    get<UnsettledPaymentResponse[]>('/monitoring/unsettled-payments'),

  getPendingSettlements: () =>
    get<PendingSettlementResponse[]>('/monitoring/pending-settlements'),

  getFinancialSummary: () =>
    get<FinancialSummaryResponse>('/monitoring/financial-summary'),
}

File: backend/app/schemas/monitoring.py
import uuid
from decimal import Decimal
from typing import Optional
from pydantic import BaseModel, ConfigDict, PlainSerializer
from typing_extensions import Annotated

DecimalStr = Annotated[Decimal, PlainSerializer(lambda x: str(x) if x is not None else None, return_type=str, when_used='json')]

class VendorLinkage(BaseModel):
    id: uuid.UUID
    vendor_code: str
    legal_name: str

    model_config = ConfigDict(from_attributes=True)

class InvoiceLinkage(BaseModel):
    id: uuid.UUID
    invoice_number: str

    model_config = ConfigDict(from_attributes=True)

class OutstandingInvoiceResponse(BaseModel):
    id: uuid.UUID
    invoice_number: str
    status: str
    total_amount: DecimalStr
    paid_amount: DecimalStr
    outstanding_amount: DecimalStr
    aging_days: int
    is_overdue: bool
    vendor: VendorLinkage

    model_config = ConfigDict(from_attributes=True)

class PendingPaymentResponse(BaseModel):
    id: uuid.UUID
    payment_reference: str
    amount: DecimalStr
    status: str
    aging_days: int
    invoice: InvoiceLinkage
    vendor: VendorLinkage

    model_config = ConfigDict(from_attributes=True)

class UnsettledPaymentResponse(BaseModel):
    id: uuid.UUID
    payment_reference: str
    amount: DecimalStr
    settlement_total: DecimalStr
    unsettled_amount: DecimalStr
    invoice: InvoiceLinkage
    vendor: VendorLinkage

    model_config = ConfigDict(from_attributes=True)

class PendingSettlementResponse(BaseModel):
    id: uuid.UUID
    settlement_reference: str
    amount: DecimalStr
    status: str
    aging_days: int
    vendor: VendorLinkage
    invoice: InvoiceLinkage

    model_config = ConfigDict(from_attributes=True)

class FinancialSummaryResponse(BaseModel):
    total_outstanding_invoices: DecimalStr
    total_pending_payments: DecimalStr
    total_unsettled_liabilities: DecimalStr
    total_pending_settlements: DecimalStr

    model_config = ConfigDict(from_attributes=True)
